### PR TITLE
Add missing profiling regions to sortByKey

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp [hunk 1]
```diff
@@ -78,6 +78,8 @@ namespace KokkosExt
 template <typename ExecutionSpace, typename Keys, typename Values>
 void sortByKey(ExecutionSpace const &space, Keys &keys, Values &values)
 {
+  Kokkos::Profiling::pushRegion("ArborX::KokkosExt::sortByKey::Kokkos");
```
Comment (dalg24) [inline src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp:81]:
```suggestion
 KokkosExt::ScopedProfileRegion guard("ArborX::KokkosExt::sortByKey::Kokkos");
```

**Review by dalg24 — CHANGES_REQUESTED** [review]
Prefer `ScopedProfileRegion`.  There is a bug in the BinSort for early returns (unbalanced push/pop regions)

**Comment by aprokop** [comment]
> Prefer ScopedProfileRegion. There is a bug in the BinSort for early returns (unbalanced push/pop regions)

Forgot it existed. And good catch.

File: src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp [hunk 2]
```diff
@@ -78,6 +79,8 @@ namespace KokkosExt
 template <typename ExecutionSpace, typename Keys, typename Values>
 void sortByKey(ExecutionSpace const &space, Keys &keys, Values &values)
 {
+  KokkosExt::ScopedProfileRegion guard("ArborX::KokkosExt::sortByKey::Kokkos");
```
Comment (dalg24) [inline src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp:82]: Would say "KokkosBinSort" but not blocking.

**Review by dalg24 — APPROVED** [review]

File: src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp (hunk 2 above)
Comment (aprokop) [reply src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp:82]: I thought about it, but thought we may switch to using Kokkos Thrust at some point in the future, so decided to leave it as generic Kokkos.

**Comment by aprokop** [comment]
retest this please

**Review by masterleinad — APPROVED** [review]

**Comment by aprokop** [comment]
retest this please

**Pull request merged**